From cf6be446cf91d0bac498ea5b41ac8d097214140a Mon Sep 17 00:00:00 2001
From: Hariprasad Kelam <hkelam@marvell.com>
Date: Fri, 24 Jul 2020 14:51:19 +0530
Subject: [PATCH 11/11] octeontx2-pf: update ethtool supported modes mask

commit 22c602bdcd1de3b215d09228f7ee45881b8a411b from
git@git.assembla.com:cavium/WindRiver.linux.git

To support Sfp&Qsfp for LIO and LIO3 new link modes are supported.
Update ethtool supported modes accordingly.

Change-Id: Ic2a7843edded4e4db29ef1556af9fefeb4eae3be
Signed-off-by: Hariprasad Kelam <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/32649
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: Just some minor context mods in order to port to linux-yocto]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
index 3095f80299ee..e3c12e5eeb81 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
@@ -47,7 +47,7 @@ struct otx2_stat {
 	.index = offsetof(struct otx2_dev_stats, stat) / sizeof(u64), \
 }
 
-#define OTX2_ETHTOOL_SUPPORTED_MODES 0x638CE23 //110001110001100111000100011
+#define OTX2_ETHTOOL_SUPPORTED_MODES 0x638CFFB //110001110001100111111111011
 #define OTX2_ETHTOOL_ALL_MODES (ULLONG_MAX)
 
 static const struct otx2_stat otx2_dev_stats[] = {
-- 
2.17.1

